rq/queue.py

```diff
@@ -310,10 +335,66 @@ def create_job(self, func, args=None, kwargs=None, timeout=None,
 
         return job
 
+    def setup_dependencies_and_enqueue(
+        self,
+        job,
+        pipeline=None,
+        at_front=False
+    ):
+        # If a _dependent_ job depends on any unfinished job, register all the
+        # _dependent_ job's dependencies instead of enqueueing it.
+        #
+        # `Job#fetch_dependencies` sets WATCH on all dependencies. If
+        # WatchError is raised in the when the pipeline is executed, that means
+        # something else has modified either the set of dependencies or the
+        # status of one of them. In this case, we simply retry.
+        if job._dependency_id is not None:
```

Comment:
Can you check with `job._dependency_ids` instead? `job._dependency_id` will be deprecated soon. I'm ok with this way of checking whether this job has dependencies.